t2box_reader/t2box_parsers_unit.pas

```diff
@@ -0,0 +1,239 @@
+unit t2box_parsers_unit;
+
+{$mode objfpc}{$H+}
+
+interface
+
+uses
+
+  Classes,
+  Dialogs,
+  Math,
+  SysUtils;
+
+type
+  TExtBytes = array [0..9] of byte;
+
+function parse_blob(var box_file: file; fldname: string; size: integer): Tbytes;
+function parse_boolean(var box_file: file; fldname: string): boolean;
+function parse_byte(var box_file: file; fldname: string): byte;
+function parse_double(var box_file: file; fldname: string): double;
+function parse_extended(var box_file: file; fldname: string): double;
+function parse_float(var box_file: file; fldname: string): single;
+function parse_integer(var box_file: file; fldname: string): integer;
+function parse_string(var box_file: file; fldname: string; strlen: integer): string;
+
+function extendedToDouble(bytes: TextBytes): double;
+function doubleToExtended(Value: double): TExtBytes;
+
+var
+  logging: boolean = false;
+
+implementation
+
+uses
+  t2box_unit;
+
+function parse_blob(var box_file: file; fldname: string; size: integer): Tbytes;
+var
+  Value: Tbytes;
+  number_read: integer;
+begin
+  setlength(Value, size);
+  BlockRead(box_file, Value, 255, number_read);
+  //need to use hexdump here
+  //if logging then
+  //  showmessage('Parsed bool : ' + fldname + ' : ' + booltostr(Value));
+  Result := Value;
+end;
+
+function parse_boolean(var box_file: file; fldname: string): boolean;
+var
+  Value: boolean;
+  number_read: integer;
+begin
+  BlockRead(box_file, Value, SizeOf(Value), number_read);
+  if logging then
+    T2box_log.Info('Parsed bool : ' + fldname + ' : ' + booltostr(Value));
+  Result := Value;
+end;
+
+function parse_byte(var box_file: file; fldname: string): byte;
+var
+  Value: byte;
+  number_read: integer;
+begin
+  BlockRead(box_file, Value, SizeOf(Value), number_read);
+  if logging then
+    T2box_log.Info('Parsed byte : ' + fldname + ' : ' + inttostr(Value));
+  Result := Value;
+end;
+
+function parse_double(var box_file: file; fldname: string): double;
+var
+  Value: double;
+  number_read: integer;
+begin
+  BlockRead(box_file, Value, SizeOf(Value), number_read);
+  if logging then
+    T2box_log.Info('Parsed double : ' + fldname + ' : ' + floattostr(Value));
+  Result := Value;
+end;
+
+function parse_extended(var box_file: file; fldname: string): double;
+var
+  bytes: TextBytes;
```

Comment:
Inconsistent caps, which affect the meaning (when reading).

Reply:
Thanks Alistair. 
Yes, as I said, badly in need of a tidying pass. 
I will pick up (hopefully) all this in the next revision.